### PR TITLE
[BD-27] [TNL-7694] [BB-3248] Support for Wiki Content in cc2olx

### PR DESCRIPTION
**Pull request by shimulch**
This PR adds support for processing canvas wiki links that defined by ``$WIKI_REFERENCE$``.

**JIRA tickets**: 
- https://openedx.atlassian.net/browse/TNL-7694
- https://tasks.opencraft.com/browse/BB-3248

~~**Discussions**:~~

**Dependencies**: None

~~**Screenshots**:~~

~~**Sandbox URL**:~~

**Merge deadline**: None

**Testing instructions**:

1. Checkout this PR.
2. Use a [canvas_cc](https://github.com/instructure/canvas_cc) common cartridge ``.imscc`` file that contains wiki pages.
3. Convert it using ``cc2olx -i <file_path>``
5. Create a new course in your local edX devstack.
4. Import output file from ``output/<file_name>.tar.gz`` via studio.
5. Check that links that had ``$WIKI_REFERENCE$`` has been replaced by ``/jump_to_id/<unit_url_name>``
6. Clicking those links from LMS redirects to the correct place.

**Author notes and concerns**:
1. To achieve this, we needed a map that points CC ``webcontent['href'] -> webcontent['identifier']`` named ``resource_id_by_href``.
2. We set OLX ``url_name`` of chapter, sequential, vertical, and unit with their respective ``identifierref`` from CC
3. When processing static URLs we try to find the appropriate unit ``url_name`` from ``resource_id_by_href`` map and replace them using edX ``/jump_to_id/<unit_id>`` pattern.

**Reviewers**
- [ ] @kaizoku
- [ ] edX reviewer[s] TBD

~~**Settings**~~

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @shimulch! I've created [BLENDED-672](https://openedx.atlassian.net/browse/BLENDED-672) to keep track of it in Jira.   More details are on the [BD-27](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1544912903/BD-27+Common+Cartridge+Converter) project page.


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: src/cc2olx/olx.py [hunk 1]
```diff
@@ -110,11 +114,37 @@ def _get_content(self, element_data):
...
+                    replace_with = '/jump_to_id/{}'.format(self.cartridge.resource_id_by_href[key])
+                    html = html.replace(item, replace_with)
+                    return html
+            logger.warn('Unable to process Wiki link - %s', item)
+            return html
+
+        for item in items:
+            if 'IMS-CC-FILEBASE' in item:
+                new_item = urllib.parse.unquote(item).replace("$IMS-CC-FILEBASE$", "/static")
+                html = html.replace(item, new_item)
```
Comment (farhaanbukhsh) [inline src/cc2olx/olx.py:144]: these two can also be moved in a function to keep things consistent

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
We do need to add test to your code, at least update the imsc_file to add the unit with wiki link and then update corresponding tests.

File: src/cc2olx/olx.py [hunk 2]
```diff
@@ -110,11 +114,37 @@ def _get_content(self, element_data):
...
+        Process static links like src and href to have appropriate links.
+        """
         srcs = re.findall(r'src\s*=\s*"(.+?)"', html)
-        for src in srcs:
-            if 'IMS-CC-FILEBASE' in src:
-                new_src = urllib.parse.unquote(src).replace("$IMS-CC-FILEBASE$", "/static")
-                html = html.replace(src, new_src)
+        hrefs = re.findall(r'href\s*=\s*"(.+?)"', html)
+
+        items = srcs + hrefs
```
[the remark this answers is not in the document]
Comment (shimulch) [reply src/cc2olx/olx.py]: Updated. Thanks for the tip :-)

File: src/cc2olx/olx.py (hunk 1 above)
Comment (shimulch) [reply src/cc2olx/olx.py]: Refactored

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested that `cc2olx with changes from this PR can convert common cartridge file and resulting course contains valid links to wiki pages. Also, I ran tests, they're passing.
- [x] I read through the code
- [x] Includes documentation

**Comment by alangsto** [comment]
@0x29a Can you fix the conflicting files? Then I can approve 👍

**Comment by shimulch** [comment]
@alangsto, conflicts are resolved and commits are squashed.

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested that issue with broken static links fixed.

**Review by alangsto — APPROVED** [review]
LGTM

**Pull request merged**